tests/pipeline/test_arrow_sources.py

```diff
@@ -610,3 +610,45 @@ def some_data():
 
         # Normalized file has _dlt_load_id
         assert pa.compute.all(pa.compute.equal(tbl["_dlt_load_id"], load_id)).as_py()
+
+
+def test_replace_existing_dlt_load_id():
```

Comment:
Another test idea: you could create a matrix with 2 dimensons: input data has this column or not on one, and setting is set to true vs false on the other, then you basically get 4 tests for free and you may be even able to remove the test above, or fuse the two.